wdl2cwl/tests/cwl_files/BuildCembaReferences.cwl

```diff
@@ -0,0 +1,395 @@
+  - id: CreateReferenceFastaIndex.ref_index_output
+    outputSource: CreateReferenceFastaIndex/ref_index_output
+    type: File
+  - id: CreateReferenceFastaIndex.monitoring_log
+    outputSource: CreateReferenceFastaIndex/monitoring_log
+    type: File
+cwlVersion: v1.2
+steps:
+  - id: Convert
+    in:
+      - id: fasta_input
+        source: Convert.fasta_input
+      - id: monitoring_script
+        source: Convert.monitoring_script
+    out:
+      - id: fwd_converted_reference_fasta_output
+      - id: rev_converted_reference_fasta_output
+      - id: monitoring_log
+    run:
+        class: CommandLineTool
+        id: BuildBisulfiteReferences
+        inputs:
+          - id: fasta_input
+            type: File
+          - id: monitoring_script
+            type:
+              - File
+              - 'null'
+        outputs:
+          - id: fwd_converted_reference_fasta_output
+            type: File
+            outputBinding:
+                glob: $("genome_mfa.CT_conversion.fa")
+          - id: rev_converted_reference_fasta_output
+            type: File
+            outputBinding:
+                glob: $("genome_mfa.GA_conversion.fa")
+          - id: monitoring_log
+            type: File
+            outputBinding:
+                glob: monitoring.log
+        requirements:
+          - class: DockerRequirement
+            dockerPull: quay.io/broadinstitute/bisulfite-references:1.0
+          - class: InitialWorkDirRequirement
+            listing:
+              - entryname: example.sh
+                entry: |4
+
+                    # if the WDL/task contains a monitoring script as input
+                    if [ ! -z "$(inputs.monitoring_script === null ? "" : inputs.monitoring_script.path)" ]; then
+                      chmod a+x $(inputs.monitoring_script === null ? "" : inputs.monitoring_script.path)
+                      $(inputs.monitoring_script === null ? "" : inputs.monitoring_script.path) > monitoring.log &
+                    else
+                      echo "No monitoring script given as input" > monitoring.log &
+                    fi
+
+                    python /build_bisulfite_references.py \
+                      --input-fasta $(inputs.fasta_input.path) \
+                      --forward-convert-out $("genome_mfa.CT_conversion.fa") \
+                      --reverse-convert-out $("genome_mfa.GA_conversion.fa")
+          - class: InlineJavascriptRequirement
+          - class: NetworkAccess
+            networkAccess: true
+          - class: ResourceRequirement
+            coresMin: 1
+            ramMin: 3337.860107421875
+            outdirMin: 1024
+        cwlVersion: v1.2
+        baseCommand:
+          - bash
+          - example.sh
+  - id: IndexForward
+    in:
+      - id: fasta_input
+        source: IndexForward.fasta_input
+      - id: index_prefix
+        source: IndexForward.index_prefix
+      - id: monitoring_script
+        source: IndexForward.monitoring_script
```

Comment:
Later steps get their inputs from the outputs of earlier steps, set defaults, or a workflow input

```suggestion
      - id: fasta_input
        source:  Convert/fwd_converted_reference_fasta_output 
        # The output named "fwd_converted_reference_fasta_output" from the step named "Convert"
      - id: index_prefix
        default: "BS_CT"
      - id: monitoring_script
        source: monitoring_script
```